tools/launch_instance.rb

```diff
@@ -889,6 +887,12 @@ def terminate(vminfo)
           # target assumed to be a service name
           iaas = iaas_by_service(target)
           iaas.terminate_by_launch_opts(spec)
+          #dns_component only used in 3.x install
+          if (not dns_component.nil?) and (not dns_component.empty?)
+            logger.info "Deleting DNS records in AWS Route53..."
+            aws_iaas = iaas_by_service("AWS-CI")
+            aws_iaas.delete_resource_records_re(/.+[.]#{dns_component}/)
```

Comment:
@jianlinliu the regexp need only be `Regexp.new(/#{spec}/)`, because in amz.rb, it's used as 
```
      to_delete = records.select { |r| r[:name] =~ re }
  ```